### ISSUE TITLE
DOM addClass, removeClass, toggleClass methods should accept whitespace

### ISSUE THREAD

**Issue by misteroneill** [videojs/video.js#7685]
## Description

During discussion on #7611, it came up that methods like `addClass` and `hasClass` will throw an error if given a value with whitespace, like `player.addClass('foo bar');`. The reason Video.js does not accept this is that the DOM `classList` object methods accept only single class names.

Specifying multiple class names through a whitespace-delimited string is a common practice in both HTML and libraries like jQuery, so we should accept them, too.

_Note:_ it does not make sense to support this for the `hasClass` method.

### Acceptance Criteria

* `hasClass` should be unchanged. It should continue to throw if whitespace is found in the class.
* `addClass`, `removeClass`, and `toggleClass` should continue functioning as today.
* `addClass`, `removeClass`, and `toggleClass` should additionally accept whitespace-delimited class names - including multiple spaces, tabs, or other whitespace characters.
* Any cases in the codebase that are calling these methods in a loop (e.g. `classes.forEach(c => dom.addClass(c))`) should be refactored in a second PR to use new method capabilities. See the change in #7611 for one example.

**Comment by Hardik-hi** [comment]
Can I work on this @misteroneill ?

**Comment by mister-ben** [comment]
@Hardik-hi are you still working on this or need help? Or could someone else take it up?

**Comment by shmruin** [comment]
@misteroneill Hi, has this issue already been resolved? I just take a look on this code, and suggest changing the code on `/src/js/utils/dom.js` like below.

```js
// Before
export function addClass(element, classToAdd) {
  if (element.classList) {
    element.classList.add(classToAdd);
  // ...
}
```

```js
// After
export function addClass(element, classToAdd) {
  if (element.classList) {
    element.classList.add(...classToAdd.split(/\s+/));
  // ...
```

And simple jsfiddle test : https://jsfiddle.net/ruin09/u7rmop9t/14/

(Maybe update this by removing line feed & carriage return from the cases, I guess)

I'm quite new here, so just ignore this if this is already resolved :)

**Comment by surmon-china** [comment]
I used the following method to handle it.

```ts
// Set new class names to Video.js container element.
const updateClassNames = (oldClassName: string | void, newClassName: string | void) => {
  // Preprocessing of className: ` test-a    test-b   test-c  ` > `test-a test-b test-c`
  const standardize = (name: string | void) => {
    const trimmed = name?.trim().replace(/\s+/g, ' ')
    return trimmed ? trimmed.split(' ') ?? [] : []
  }

  standardize(oldClassName).map((name) => player.removeClass(name))
  standardize(newClassName).map((name) => player.addClass(name))
}
```

**Comment by gkatsev** [comment]
Fixed via #7798
